challenges/objects-arrays.js

```diff
@@ -85,35 +129,64 @@ zooAnimals = [{"animal_name":"Jackal, asiatic","population":5,"scientific_name":
 
 /* Request 1: .forEach()
 
-The zoo wants to display both the scientific name and the animal name in front of the habitats.  Return an array with only the animal and scientific names in it.  The individual values in the array should look like this "Name: Jackal, asiatic, Scientific: Canis aureus."
+The zoo wants to display both the scientific name and the animal name in front of the habitats.  
+Return an array with only the animal and scientific names in it.  
+The individual values in the array should look like this "Name: Jackal, asiatic, Scientific: Canis aureus."
 
 */
-const animalNames = [];
+
+
+const animalNames = []
+
+zooAnimals.forEach(name)
+
+function name (scien) {
+  animalNames.push (`Name: ${scien.animal_name}, Scientific: ${scien.scientific_name}.`)
+}
 console.log(animalNames);
 
 /* Request 2: .map()    
 
-The zoos need a list of all their animal's names (names only, not scientific) converted to lower case.  Create a new array named lowerCase and map over each name to convert them all to lower case.  Log the resut.
+The zoos need a list of all their animal's names (names only, not scientific) converted to lower case.  
+Create a new array named lowerCase and map over each name to convert them all to lower case.  Log the resut.
 
 */
 
-const lowerCase = [];
+const lowerCase = zooAnimals.map(lower)
+
+function lower (name) {
+  return name.animal_name.toLowerCase()
+}
 console.log(lowerCase); 
 
 /* Request 3: .filter() 
 
 The zoos are concenred about animals with a lower population count. Find out which animals have a population less than 5.
 
 */
-const largerPopulation = [];
+const largerPopulation = []
+
+zooAnimals.filter(populations)
+
+function populations (pop) {
+  if (pop.population < 5) {
+  largerPopulation.push (pop.animal_name)
+  }
```

Comment:
So the callback you are passing `.filter` isn't working properly. Filter like `map` returns an array, but it only adds items that the argument it receives evaluates down to true. That is why we have the call back function passed in as the `filter` argument.

Meaning the callback function passed into the `filter` MUST evaluate to `true` or `false`. So really here all you actually need to `return`, without the if conditional is simply `pop.population > 5`and instead of in setting the `largerPopulation` to an empty array it would be `const largerPopulation = zooAnimals.filter(populations)`